src/StorageManager.cpp

```diff
@@ -24,6 +24,12 @@
 
 using namespace jni;
 
+CJNIStorageVolume CJNIStorageManager::getPrimaryStorageVolume()
+{
+  return call_method<jhobject>(m_object, "getPrimaryStorageVolume",
+    "()Landroid/os/storage/StorageVolume;");
+}
```

Comment:
If the other one should be gated, i guess the theory this should be as well (API 24)?. There "may" be a time where libandroidjni is wanted to be backported to something other than master, its just a lib after all. At the very least currently, Omega still has 21 as a minimum.

Reply:
It's an option, but I've followed the general criteria of the library, which is not to include the level check in the function, leaving it to the app to control this, as it may require an alternative implementation if it can execute at a lower level.

If libandroidjni is backported there is no issue because it includes the old and the new functions.